src/Microsoft.AspNet.Authentication.Google/Notifications/GoogleAuthenticationNotifications.cs

```diff
@@ -23,16 +24,33 @@ public GoogleAuthenticationNotifications()
         /// <summary>
         /// Gets or sets the function that is invoked when the Authenticated method is invoked.
         /// </summary>
-        public Func<GoogleAuthenticatedContext, Task> OnAuthenticated { get; set; }
+        public new Func<GoogleAuthenticatedContext, Task> OnAuthenticated { get; set; }
 
         /// <summary>
         /// Invoked whenever Google succesfully authenticates a user.
         /// </summary>
-        /// <param name="context">Contains information about the login session as well as the user <see cref="System.Security.Claims.ClaimsIdentity"/>.</param>
+        /// <param name="context">Contains information about the login session as well as the user <see cref="ClaimsIdentity"/>.</param>
         /// <returns>A <see cref="Task"/> representing the completed operation.</returns>
         public virtual Task Authenticated(GoogleAuthenticatedContext context)
         {
             return OnAuthenticated(context);
         }
+
+        /// <summary>
+        /// Invoked after the provider successfully authenticates a user.
+        /// </summary>
+        /// <param name="context">Contains information about the login session as well as the user <see cref="ClaimsIdentity"/>.</param>
+        /// <returns>A <see cref="Task"/> representing the completed operation.</returns>
+        public sealed override async Task Authenticated(OAuthAuthenticatedContext context)
+        {
+            var notification = context as GoogleAuthenticatedContext;
+            if (notification == null)
+            {
+                throw new InvalidOperationException("OAuthAuthenticationHandler.CreateTicketAsync must be overridden " +
```

Comment:
Ok, but that only make sense if you copy pasted this method into your own notification class. I don't think we should add an exception that only fires for developer copy-paste errors. At most this would be a Debug.Assert.